model/src/main/scala/org/zalando/nakadi/client/model/Model.scala

```diff
@@ -0,0 +1,192 @@
+ * @param parentEids Event identifier of the Event that caused the generation of this Event. Set by the producer.
+ * @param flowId The flow-id of the producer of this Event. As this is usually a HTTP header, this is enriched from the header into the metadata by Nakadi to avoid clients having to explicitly copy this.
+ * @param metadata This Metadata contains common fields unrelated to Nakadi logic. Should be mainly enriched by the Consumer.
+ *
+ */
+case class EventMetadata(
+  eid: String,
+  eventType: String,
+  occurredAt: String,
+  receivedAt: String,
+  parentEids: Seq[String],
+  flowId: String,
+  metadata: Map[String, Any])
+
+/**
+ * A Business Event. Usually represents a status transition in a Business process.
+ *
+ * @param metadata Metadata for this Event. Contains commons fields for both Business and DataChange Events. Most are enriched by Nakadi upon reception, but they in general MIGHT be set by the client.
+ */
+case class BusinessEvent(metadata: EventMetadata)
+
+/**
+ * Indicators of a `DataChangeEvent`'s referred data type and the type of operations done on them.
+ * @param dataType The datatype of the `DataChangeEvent`.
+ * @param dataOp The type of operation executed on the entity. * C: Creation * U: Update * D: Deletion * S: Snapshot
+ */
+case class DataChangeEventQualifier(
+  dataType: String,
+  dataOperation: DataOperation.Value)
+
+/**
+ * A Data change Event. Represents a change on a resource.
+ *
+ * @param data The payload of the type
+ * @param eventQualifier Indicators of a `DataChangeEvent`'s referred data type and the type of operations done on them.
+ * @param metadata Metadata for this Event. Contains commons fields for both Business and DataChange Events. Most are enriched by Nakadi upon reception, but they in general MIGHT be set by the client
+ */
+case class DataChangeEvent[T](
+  data: T,
+  eventQualifier: DataChangeEventQualifier,
+  metadata: EventMetadata)
+
+/**
+ * @ param problemType An absolute URI that identifies the problem type. When dereferenced, it SHOULD provide human-readable API documentation for the problem type (e.g., using HTML). This Problem object is the same as provided by https://github.com/zalando/problem
+ * @ param title A short, summary of the problem type. Written in English and readable for engineers (usually not suited for non technical stakeholders and not localized)
+ * @ param status The HTTP status code generated by the origin server for this occurrence of the problem.
+ * @ param detail A human readable explanation specific to this occurrence of the problem.
+ * @ param instance An absolute URI that identifies the specific occurrence of the problem. It may or may not yield further information if dereferenced.
+ */
+case class Problem(
+  problemType: String,
+  title: String,
+  status: Int,
+  detail: String,
+  instance: String)
+
+case class Metrics(metrics:String) //TODO: It is not defined yet!
```

Comment:
there is no fixed structure. see recent discussion: metrics can be a map of maps with maps... + there are some string leafs in between